# Hypothesis testing problems

## Exercise 1

You are a nutritionist researching two different types of diets to see if there is a significant difference in weight loss after one month. You choose two random groups of people; one group follows the first diet, and the other follows the second. At the end of the month, you record the weight loss (in kg) for each person:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

With these data, it seeks to answer the following question: Is there a significant difference in average weight loss between people who followed the first diet and those who followed the second diet?

To draw conclusions, follow the points below:

- State the hypothesis: null and alternative hypothesis.
- Perform the test to test the hypothesis. You can use a Student's t-test.
- Analyze the conclusions.

In [1]:
import numpy as np
from scipy.stats import ttest_ind

# Data for Diet 1 and Diet 2
diet1 = [2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5]
diet2 = [3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2]

# Perform independent t-test
t_stat, p_value = ttest_ind(diet1, diet2, equal_var=True)

t_stat, p_value


(np.float64(-3.5383407969933938), np.float64(0.004083270191713912))

Analyze the Conclusions
T-Statistic: -3.54
P-Value: 0.00408


The p-value (0.00408) is less than 0.05, indicating that we reject the null hypothesis 

The conclusion is that there is a significant difference in the average weight loss between Diet 1 and Diet 2.
The  analyze is displaying that the two diets produce significantly different average weight loss results after one month. ​

## ANOVA

**ANOVA** (*Analysis of Variance*) is a statistical technique used to compare the measures of two or more groups. The idea behind ANOVA is to decompose the total variability in the data into two components: between-group variability and within-group variability:

- **Between-group variability**: This variability refers to the differences between the group means. If this variability is considerably larger than the within-group variability, it could be an indication that at least one of the group means is different.
- **Within-group variability**: This variability refers to the dispersion of the data within each group. If all groups have similar variability, then any noticeable difference in group means could be considered significant.

Hypotheses in ANOVA typically include:

- **Null hypothesis** ($H₀$): The means of all groups are equal.
- **Alternative hypothesis** ($H₁$): At least one of the group means is different.

If the ANOVA test result is significant (e.g., a p-value less than a threshold such as 0.05), this suggests that at least one group mean is different.

## Exercise 2

A farmer decides to test three different types of fertilizers to determine if one is superior in terms of corn production. The farmer plants corn on 15 identical plots and uses all three fertilizers (5 plots for each type). At the end of the season, he measures the corn yield (in kg) of each plot, with the following result:

| Fertilizer 1 | Fertilizer 2 | Fertilizer 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

With this data, he seeks to answer the following question: Is there a significant difference in average corn yield between the three types of fertilizers?

To help you, follow the points below:

- State the hypothesis: null and alternative hypothesis.
- Perform the ANOVA test.
- Analyze the conclusions.
- If one fertilizer is better than another, how can we know it?

In [2]:
from scipy.stats import f_oneway

# Data for each fertilizer
fertilizer1 = [20, 21, 20, 19, 20]
fertilizer2 = [22, 21, 23, 22, 21]
fertilizer3 = [24, 23, 22, 23, 24]

# Perform one-way ANOVA
f_stat, p_value = f_oneway(fertilizer1, fertilizer2, fertilizer3)

f_stat, p_value


(np.float64(20.315789473684188), np.float64(0.00014047824793190475))

F-Statistic: 20.32
P-Value: 0.00014

The p-value (0.00014) is much smaller than 0.05, so we reject the null hypothesis.
This means there is a statistically significant difference in the average corn yield between at least two of the fertilizer types.

In [3]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd
import pandas as pd

# Combine data into a single list and create corresponding labels
data = fertilizer1 + fertilizer2 + fertilizer3
labels = ['Fertilizer 1'] * len(fertilizer1) + ['Fertilizer 2'] * len(fertilizer2) + ['Fertilizer 3'] * len(fertilizer3)

# Perform Tukey's HSD test
tukey_results = pairwise_tukeyhsd(data, labels, alpha=0.05)

# Display results
tukey_results.summary()


<class 'statsmodels.iolib.table.SimpleTable'>

The Tukey's HSD test shows significant differences in average corn yield between all pairs of fertilizers. Fertilizer 3 produces a significantly higher yield compared to both Fertilizer 1 and Fertilizer 2, suggesting that it is the most effective among the three. 